server/src/server.ts

```diff
@@ -169,6 +169,15 @@ const analyzeDocument = async (event: TextDocumentChangeEvent<TextDocument>): Pr
 
 documents.onDidOpen(analyzeDocument)
 
-documents.onDidChangeContent(analyzeDocument)
+documents.onDidChangeContent(async (event: TextDocumentChangeEvent<TextDocument>) => {
+  await analyzeDocument(event)
+
+  const possibleSrcUris = await connection.sendRequest<RequestResult['getPossibleSrcUris']>(RequestMethod.getPossibleSrcUris, { uri: event.document.uri.replace('file://', '') })

```

Comment:
Has a significant CPU cost but it's not unreasonable.

Reply:
I now cache the data, I will implement a way to clear the cache when a file is renamed, created and deleted. But I want to do it after the https://github.com/yoctoproject/vscode-bitbake/pull/206 to prevent conflicts.

Reply:
Managing such cache can be tricky, I think it's fine to keep it as it was in this PR.
I can't identify the cache logic properly in this commit stack. THere seem to be no cleanup, no update on file manipulation, ...
It's also optional to have a cache for this at all IMO. Except if you could produce real CPU hog with large folders?

Reply:
Now it stores the local file info in the `analyzer` and it clears all info upon file operations.